examples/generative/molecule_generation.py

```diff
@@ -0,0 +1,493 @@
+data-driven continuous representation of molecules**](https://arxiv.org/abs/1610.02415) as a reference.
+The model explained in this paper generate new molecules for efficient exploration and
+optimization through open-ended spaces of chemical compounds. The Model consists of three
+components: Encoder, Decoder and Predictor.  The encoder converts the discrete
+representation of a molecule into a real-valued continuous vector, and the decoder
+converts these continuous vectors back to discrete molecular representations. The
+predictor estimates chemical properties from the latent continuous vector representation
+of the molecule. Continuous representations allow the use of powerful gradient-based
+optimization to efficiently guide the search for optimized functional compounds.
+
+![intro](https://pubs.acs.org/na101/home/literatum/publisher/achs/journals/content/acscii/2018/acscii.2018.4.issue-2/acscentsci.7b00572/20180223/images/medium/oc-2017-00572f_0001.gif)
+
+**Figure (a)** - A diagram of the autoencoder used for molecular design, including the
+joint property prediction model. Starting from a discrete molecular representation, such
+as a SMILES string, the encoder network converts each molecule into a vector in the
+latent space, which is effectively a continuous molecular representation. Given a point
+in the latent space, the decoder network produces a corresponding SMILES string. A
+mutilayer perceptron network estimates the value of target properties associated with
+each molecule.
+
+**Figure (b)** - Gradient-based optimization in continuous latent space. After training a
+surrogate model f(z) to predict the properties of molecules based on their latent
+representation z, we can optimize f(z) with respect to z to find new latent
+representations expected to have high values of desired properties. These new latent
+representations can then be decoded into SMILES strings, at which point their properties
+can be tested empirically.
+"""
+
+"""
+## Dataset
+
+We would be using [**ZINC – A Free Database of Commercially Available Compounds for
+Virtual Screening**](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1360656/) dataset. The
+dataset comes with molecular formula in SMILE representation along with their respective
+molecular properties such as **logP**( Shows the water–octanal partition coefficient ),
+**SAS**( Shows the synthetic accessibility score ) and **QED**( Shows the Qualitative
+Estimate of Drug-likeness ).
+
+"""
+
+"""shell
+!wget
+https://raw.githubusercontent.com/aspuru-guzik-group/chemical_vae/master/models/zinc_properties/250k_rndm_zinc_drugs_clean_3.csv
+"""
+
+"""
+## Setup
+"""
+
+"""shell
+!pip -q install rdkit-pypi
+"""
+
+import ast
+
+import pandas as pd
+import numpy as np
+
+from rdkit import Chem
+from rdkit import RDLogger
+from rdkit.Chem.Draw import IPythonConsole
+from rdkit.Chem.Draw import MolsToGridImage
+
+import tensorflow as tf
```

Comment:
Import Keras too (`from tensorflow import keras`) and use it instead of `tf.keras`, that will save you some space